include/vara/Feature/FeatureModelTransaction.h

```diff
@@ -326,6 +342,46 @@ class AddFeatureToModel : public FeatureModelModification {
   Feature *Parent;
 };
 
+//===----------------------------------------------------------------------===//
+//                       RemoveConstraintFromModel
+//===----------------------------------------------------------------------===//
+
+class RemoveConstraintFromModel : public FeatureModelModification {
+  friend class FeatureModelModification;
+  friend class RemoveFeatureFromModel;
+
+public:
+  Result<FTErrorCode> exec(FeatureModel &FM) override {
+    if (auto E = (*this)(FM); !E) {
+      return E.getError();
+    }
+    return Ok();
+  }
+
+  Result<FTErrorCode> operator()(FeatureModel &FM) {
+    // TODO: Any error handling needed here?
+    assert(RemoveConstraint.getParent() == nullptr &&
+           "Subtree deletion is not supported");
```

Comment:
We should talk about this today -- as far as I understand, uncoupling currently only works for this case.